crates/biome_js_analyze/src/lint/nursery/no_useless_string_raw.rs

```diff
@@ -0,0 +1,113 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsTemplateElement, JsTemplateExpression};
+use biome_rowan::{AstNode, AstNodeList};
+
+declare_lint_rule! {
+    /// Disallow unnecessary `String.raw` function in template string literals without any escape sequence.
+    ///
+    /// `String.raw` is useless when contains a raw string without any escape-like sequence.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// String.raw`a`;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// String.raw`a ${v}`;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// String.raw`\n ${a}`;
+    /// ```
+    ///
+    /// ```js
+    /// String.raw`\n`;
+    /// ```
+    pub NoUselessStringRaw {
+        version: "next",
+        name: "noUselessStringRaw",
+        language: "js",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessStringRaw {
+    type Query = Ast<JsTemplateExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let tag = node.tag()?;
+        let tag = tag.as_js_static_member_expression()?;
+
+        let object = tag.object().ok()?;
+        let object_expr = object.as_js_identifier_expression()?;
+        let object_name = object_expr.name().ok()?.text();
```

Comment:
Calling `text()` on a node allocates a string.
We could avoid it by retrieving the token and the calling `text_trimmed_text()`
```suggestion
        let object_name = object_expr.name().ok()?.names().ok()?.value_token().ok()?;
        let object_name = object_name.text_trimmed_text();
```

Reply:
updated~ maybe you mean `text_trimmed()` here.